src/renderer/components/+preferences/application.tsx

```diff
@@ -49,7 +52,8 @@ export const Application = observer(() => {
         : "System default shell"
     );
 
-  const [shell, setShell] = React.useState(UserStore.getInstance().shell || "");
+  const [shell, setShell] = React.useState(userStore.shell || "");
```

Comment:
In general, I have a bit of an issue with using React state and MobX state wantonly, but let's make a bigger deal of that somewhere else, some other time.